Config-Fix-Doc-Update (#1)
Jomik/screeps-server: README.md

Mods and bots are managed at startup by checking your `config.yml`.
`npm` is only invoked if changes are made to your `config.yml`.

## Installation (Recommended)

> **Step 1: Clone/Download this repository.**

Reviewer: Maybe it would be better to just expose the `docker-compose.yml` and `config.yml` in a release.
But those are the only two files you need, so unsure if the best is to clone this repo. It is of course tiny.
I personally have a docker-compose and a config in my screeps bot repo : https://github.com/Jomik/screeps-ai

Author: We would need to expose the `.env` as well. I think this way is probably "easiest" per-se.

**Step 2:  Download Docker Desktop** 

> - [Download For Mac (Intel)](https://desktop.docker.com/mac/main/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=dd-smartbutton&utm_location=module) 
> - [Download For Mac (Apple Silicon)](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=dd-smartbutton&utm_location=module)
> - [Download For Windows](https://desktop.docker.com/win/main/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=dd-smartbutton&utm_location=module)
> - [Download For Linux](https://docs.docker.com/desktop/linux/install/)

Reviewer: I would prefer just linking to Docker Desktop, and let their site handle redirecting the user to the correct thing.

```suggestion
- [Download](https://www.docker.com/products/docker-desktop/)
```

**Step 3: Open your download directory `.../screeps-server/`** 

**Step 4: Edit `config.yml`**

- Add your username to the `whitelist:`
- Adjust other settings to your likings.
	- [Docs: screepsmod-auth](https://github.com/ScreepsMods/screepsmod-auth)

> 	- [Docs: screepsmod-admin-utils](https://github.com/ScreepsMods/screepsmod-admin-utils)

Reviewer: This actually is not currently supported - I have not found a need for it.
If we want this, then we need to decide on the best way to expose this ability :)

> **Step 5: Use Docker to setup an instance.** 

Reviewer: We need to ensure that the user has a `.env` with their `STEAM_KEY`.

> - Windows
> 	- Open Windows Powershell
> 	- Navigate to your `screeps-server` directory. 
> 		- `cd /screeps-server/`
> 	- Run `docker compose up`
> 	- To open CLI run `docker compose exec screeps cli`
> - Linux & Apple
> 	- *Not added yet.*

Reviewer: The steps are essentially the same.
Maybe we should avoid this absolute path in the `cd` step.
```suggestion
	- Open a terminal
	- Navigate to your `screeps-server` directory. 
		- `cd /path/to/screeps-server/`
	- Run `docker compose up` (Press `CTRL + C` to stop the server again)
	- To open CLI run `docker compose exec screeps cli`
```
You can also use `docker compose up -d` to run it in the background. Then you would use `docker compose stop` or `docker compose down`.

## Usage
**Launching The Server**

- ***Via Docker Desktop***
	
	- If you installed everything properly using the recommended settings above, you can simply launch `Docker Desktop` and you'll see an option there for `screeps-server` which contains instances of screeps, mongo, and redis. Simply hit the `Play/Start` button on the `screeps_server` section.
	
- ***Via Docker Cli***
	
	- Windows Powershell
		- Navigate to your `screeps-server` directory.
		- Run `docker compose up`
		
	- Terminal (Linux & Mac)

> 		- *Not added yet.*

Reviewer: This is the same on Linux & Mac.

**Launching The CLI**

- Windows Powershell
  - Navigate to your `screeps-server` directory.
  - Run `docker compose exec screeps cli`
- Terminal (Linux & Mac)

>   - *Not added yet.*

Reviewer: This is the same on all platforms.

  overmind: screeps-bot-overmind
```

## Troubleshooting
- Help, my server is running but I can't connect.
  - Follow the instructions for [screepsmod-auth](https://github.com/ScreepsMods/screepsmod-auth)
- I can't push any code via `rollup` to my server.
  - Make sure your `screeps.json` configuration in your project is set properly. 
  - In your `email:` field, simply put in your `username`. Verify your password is the same as your `screepsmod-auth` setting.

- My map is all red, I can't actually spawn in!
  - Restart your server, check your configuration and follow the instructions for [screepsmod-admin-utils](https://github.com/ScreepsMods/screepsmod-admin-utils)

>   - Ensure your username is set to the `whitelist:` setting in `config.yml`

Reviewer: This is not supported. There is no whitelisting by default.

> ## Resources & Docs

Reviewer: ```suggestion
```
This is actually not visible in the rendered markdown, it just lets us write `[screepsmod-auth]` and then it will automatically create the link :)
